pkg/registry/core/pod/storage/eviction.go

```diff
@@ -154,6 +154,22 @@ func (r *EvictionREST) Create(ctx context.Context, obj runtime.Object, createVal
 		pdb := &pdbs[0]
 		pdbName = pdb.Name
 		refresh := false
+		// Pending Pods don't count towards healthy; allow removal of pending pod as long as we have enough healthy ones.
+		if pod.Status.Phase == api.PodPending && pdb.Spec.MaxUnavailable != nil {
```

Comment:
Is the bug in this case a pod that hasn’t been scheduled yet?  A pod that hasn’t been scheduled can absolutely no-questions-asked be evicted, and I don’t see that handled up above in this file.  If the bug is scheduled but not started, I’d probably still like a section above for it.

Is there any scenario where a pending pod being marked deleted could violate the constraints PDB seeks to maintain?  Ie due to a delay in propagating status?  I don’t think there is but I want to sure you’ve thought through it.

Reply:
I think I agree with this. If the pod is in a pending state, we don't really need to care about PDBs at all since the pod is not part of the PDB calculations. Then we should be able to handle this case above with the deletion of terminal pods.

Could there be a race where a pod reaches the running phase and is observed by the disruption controller before it gets deleted here? It does sound unlikely, but I think it might be possible.

Reply:
So I am not very concerned about the possible race here. It seems like there will always be a chance that reality changes between the check in the controller and the delete of a pod.

I'm not convinced the extra checks here are necessary. If we determine that the pod is in the pending state, I think we should be able to delete it without any additional checks since pending pods are not part of the pdb calculation. This means we should be able to just add a check for pending pods on line 126. Do you have any concerns about doing it that way?

Reply:
@mortent The only thing I can really think of is that if PDBs are already below tolerance, then only checking for pending state could potentially prolong getting to the right number of replicas.  While not ideal, the application is no worse off than it was.

I'll cut a new patch for just checking on 126.